tools/c7n_mailer/c7n_mailer/msg-templates/slack_default.j2

```diff
@@ -33,5 +33,6 @@
    {%- if not recipient.startswith('https://') %}
    "channel":"{{ recipient }}",
    {%- endif -%}
-   "username":"Custodian"
+   "username":"Custodian" {#This line determines which message system private messages are sent through#}
```

Comment:
if `username` is not specified, it uses the slack app?

Reply:
yes, so in slack there is the app we have made called custodian. For private messaging if the user name is defined the message will come through Slackbot. If it is left undefined it comes though a slack app just like the google calendar sack app.